xc/device.ml

```diff
@@ -1976,7 +1969,14 @@ module Backend = struct
           ~timeout ~cancel ();
         pid
 
-      let stop        = Dm_Common.stop
+      let stop ~xs ~qemu_domid domid =
+        let pid = Qemu.pid ~xs domid in
+        Dm_Common.stop ~xs ~qemu_domid domid;
+        match pid with
+        | None -> () (* nothing to do *)
+        | Some qemu_pid ->
+          Generic.best_effort "removing core files from /var/xen/qemu"
+            (fun () -> Unix.rmdir ("/var/xen/qemu/"^(string_of_int qemu_pid)))
```

Comment:
Why not using `Xenops_utils.Unix.rm` here as well?

Reply:
To retain core files if they exist. I am not sure that this still can contain core files but this was the logic.